pkg/watcher/watcher.go

```diff
@@ -97,17 +101,26 @@ func containerIndexFunc(obj interface{}) ([]string, error) {
 	return nil, fmt.Errorf("%w - found %T", errNoPod, obj)
 }
 
+// podAddrIndexFunc index pod by pod IP.
+func podAddrIndexFunc(obj interface{}) ([]string, error) {
+	switch t := obj.(type) {
+	case *corev1.Pod:
+		return []string{t.Status.PodIP}, nil
+	}
+	return nil, fmt.Errorf("%w - found %T", errNoPod, obj)
+}
+
 // NewK8sWatcher returns a pointer to an initialized K8sWatcher struct.
 func NewK8sWatcher(k8sClient *kubernetes.Clientset, stateSyncIntervalSec time.Duration) *K8sWatcher {
-	k8sInformerFactory := informers.NewSharedInformerFactoryWithOptions(k8sClient, stateSyncIntervalSec,
```

Comment:
I think it will make sense to enable this feature with feature flags (disabled by default). Also, later I would like to extend pod info with nodes zones to track multi-zone traffic. Such mapping can be also enabled by flag as it will require nodes indexer.

Reply:
Ah, yes, I see there is `ciliumState.GetIPCache().GetIPIdentity()` which returns pod and namespace. If option.Config.EnableCilium is enabled we could use this one instead of adding new indexer.